misk-admin/api/misk-admin.api

```diff
@@ -339,7 +339,7 @@ public final class misk/web/dashboard/DashboardTabLoaderEntry {
 	public fun toString ()Ljava/lang/String;
 }
 
-public final class misk/web/dashboard/DashboardTabProvider : javax/inject/Provider {
+public final class misk/web/dashboard/DashboardTabProvider : misk/web/dashboard/ValidWebEntry, javax/inject/Provider {
```

Comment:
Would you mind explaining how this solve the issue? I'm just not familiar enough with `ValidWebEntry` . The PR looks good but I'm a bit uncertain how it resolved our weird spinner issue

Reply:
`ValidWebEntry` enforces that the `DashboardTabProvider` has been constructed with a valid slug and URL path prefix. This currently happens later in the lifecycle when the  `DashboardTabProvider` creates the `DashboardTab` on `get()` but that error is opaque and was difficult to debug.

Reply:
The weird spinner issue was from an endpoint failing with the above invalid slug error. The change to the `DashboardModule` fixed the `createMenuLink` to ensure that the slug is always valid.